kyverno-pss-policy-exceptions/README.md

```diff
@@ -0,0 +1,167 @@
+
+Multiple alternatives have been considered and attempted over the past two years, including:
+
+### Exceptions via Labels / Annotations
+
+This approach allows resources to opt themselves out of policies by applying certain labels or annotations to the resource.
+
+The advantage of this approach was that it made self-service very easy for developers, and was for some time the only technically feasible way to handle self-service exceptions in Kyverno logic.
+
+There are several major drawbacks, the largest being that labels and annotations are not RBAC-controlled, so this allows most users the ability to exempt themselves from any policy without administrator review.
+
+### Exceptions via Policy
+
+This method captures all exceptions to a policy inside the policy itself. When a new exception is approved, it is added to the Kyverno `Policy` or `ClusterPolicy` resource.
+
+The advantage here was that there was a true single source of truth about the expected policy behavior.
+
+The disadvantage was virtually everything else.
+Under this model, only a centralized exception model could be adopted because an administrator would need to add and re-deploy a policy for each new exception. It requires constant modification of and addition to the policy resource, which would eventually hit the etcd resource size limit in complex environments.
+It also made it much more difficult to continue using official upstream policies - each policy would need to be copied and heavily modified both by customers and by Giant Swarm.
+This would be very error-prone and simply would not scale.
+
+## References
+
+- [Intranet page](https://intranet.giantswarm.io/docs/dev-and-releng/psp-deprecation/) about PSP to Kyverno migration and [corresponding public blog post](https://www.giantswarm.io/blog/giant-swarms-farewell-to-psp)
+- [Sample `PolicyException` resource for falco-app](https://github.com/giantswarm/falco-app/blob/main/helm/falco/templates/falco-policy-exception.yaml)
+- [Kyverno PolicyException documentation](https://kyverno.io/docs/writing-policies/exceptions/)
+
+## Appendix
+
+### Examples
+
+#### Centralized Exceptions
+
+Most customers prefer to centralize the approval and management of policy exceptions. This has historically included policies of several types, including Pod Security Policies and Network Policies, among others.
+These organizations already have established approval processes, in some cases including automated portals where users can request an exception to be automatically deployed to clusters after review by cluster administrators and/or a security team.
+
+Customers operating under this model may prefer to restrict PolicyException creation only to cluster administrators and approved automation.
+Exceptions might be stored in a single repository and/or deployed in a single namespace for simplicity.
+
+In the proposed setup, this can be achieved by:
+
+- Not granting the `CREATE polex` permission to non-administrator entities, and
```

Comment:
dumb - `polex` is short for `PolicyException`, right?

Reply:
Correct